test/task_test.rb

```diff
@@ -47,25 +39,25 @@ def iodrivers_base_prepare(model)
             0x38
         ]
 
-        syskit_wait_ready(writer, component: task)
-        sample = expect_execution do
-                     writer.write Types.iodrivers_base.RawPacket.new(
-                         time: Time.now, data: received_frame
-                     )
-                 end
-                 .to { have_one_new_sample task.generator_state_port }
+        sample = assert_driver_processes_frame(frame, task.generator_state_port)
 
-        assert_equal (((2 * Math::PI) / 60) * ((0x01 << 8) | 0x00)).round(5),
-                     sample.rotation_speed.round(5)
-        assert_equal (0.01 * ((0x03 << 8) | 0x02)).round(5),
-                     sample.start_battery_voltage.round(5)
+        assert_equal(
```

Comment:
There is `assert_in_delta` for this. `assert_in_delta x, y, precision`. You usually can omit the precision (the default is good for a lot of common cases)

Every test harness / framework has special assertions for floating-point comparison.